### PR TITLE
chore: bump react.js from 16 to 18

### PR DESCRIPTION
**Pull request by JeffreyMa597**
What problem does this PR solve?
------
Since React.js 16.8, there exists a few important changes, the latest version of react.js is v18.2.0. So it seems necessary for us to update our React version.

Updating the react version may involve a number of other dependencies that need to be upgraded simultaneously,
such as react-dom, react-script, node, typescript, etc. I'll update the version to 17 first, and then up to 18 if possible.

Ref: https://github.com/Magickbase/neuron-public-issues/issues/73

Check List
------

### Test
e2e Test
### Task
none

### CODE REVIEW

**Comment by Cedar67** [comment]
bootstrap error.

File: packages/neuron-ui/src/components/DataSetting/hooks.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export const useDataPath = (
     })
   }, [closeDialog, type])
   const onConfirm = useCallback(
-    e => {
+    (e: any) => {
       const { dataset } = e.currentTarget
```
Comment (yanguoyu) [inline packages/neuron-ui/src/components/DataSetting/hooks.ts:56]: Why use any to declare `e`?

File: packages/neuron-ui/src/widgets/Dropdown/index.tsx [hunk 2]
```diff
@@ -68,7 +68,7 @@ export const CustomizableDropdown = ({
   )
   return (
     <div className={`${styles.customizableRoot} ${className || ''}`}>
-      {children}
+      <div>{children}</div>
```
Comment (Keith-CY) [inline packages/neuron-ui/src/widgets/Dropdown/index.tsx:71]: Is wrapping `children` in `div` related to the upgrade of react?

Comment (JeffreyMa597) [reply packages/neuron-ui/src/widgets/Dropdown/index.tsx:71]: > Is wrapping `children` in `div` related to the upgrade of react?

Yes, it is related to the upgrade of react 18, in React 17 it's necessary. I'll change it back.

File: packages/neuron-ui/src/components/DataSetting/hooks.ts (hunk 1 above)
Comment (JeffreyMa597) [reply packages/neuron-ui/src/components/DataSetting/hooks.ts:56]: > Why use any to declare `e`?

Cause some of the parameters are vaguely defined, such as `document.addEventListener('click', e =>{ ... })`, I'll make a double check to remove the `any` type as much as possible.

File: packages/neuron-ui/src/components/LockInfoDialog/index.tsx [hunk 3]
```diff
@@ -55,7 +55,7 @@ const LockInfoDialog = ({ lockInfo, isMainnet, onDismiss }: LockInfoDialogProps)
 
   const onDialogClicked = useCallback(
     (e: any) => {
-      if (e.target.tagName === 'DIALOG') {
+      if (e.target.tagName === ' DIALOG') {
```
Comment (Keith-CY) [inline packages/neuron-ui/src/components/LockInfoDialog/index.tsx:58]: This could be a typo

Comment (JeffreyMa597) [reply packages/neuron-ui/src/components/LockInfoDialog/index.tsx]: > This could be a typo

@Keith-CY Fixed.

**Review by Keith-CY — APPROVED** [review]

**Review by yanguoyu — APPROVED** [review]

**Comment by Keith-CY** [comment]
Conflicted

**Comment by Cedar67** [comment]
Verified on Windows OS, no exception found now.
Regression test is blocked on macOS, because of a blocking error at startup.

**Comment by Cedar67** [comment]
> 



> Verified on Windows OS, no exception found now. Regression test is blocked on macOS, because of a blocking error at startup.

Verified on macOS, no exception found now.

**Review by Cedar67 — APPROVED** [review]

**Pull request merged**